### PR TITLE
Bluetooth: Mesh: Fix shell model clients

### PR DESCRIPTION
**Pull request by akredalen**
* Updated shell clients to comply with new bt mesh model structure.
* Small fix in shell_sensor_cli.c with array subscript causing a warning
* Updated test coverage in tests/subsys/bluetooth/mesh/models.
  Added app overlay for Distance Measurement module.

### CODE REVIEW

File: tests/subsys/bluetooth/mesh/models/testcase.yaml [hunk 1]
```diff
@@ -18,47 +18,12 @@ tests:
       - CONFIG_BT_SETTINGS=y
       - CONFIG_BT_MESH_SCENE_SRV=y
       - CONFIG_BT_MESH_SCHEDULER_SRV=y
+  bluetooth.mesh.build_models.shell:
+    extra_configs:
+      - OVERLAY_CONFIG=overlay-mesh-shell.conf
```
Comment (m-alperen-sener) [inline tests/subsys/bluetooth/mesh/models/testcase.yaml:23]:
```suggestion
    extra_configs: OVERLAY_CONFIG=overlay-mesh-shell.conf
```

File: tests/subsys/bluetooth/mesh/models/testcase.yaml [hunk 2]
```diff
@@ -18,47 +18,12 @@ tests:
...
   bluetooth.mesh.build_models.emds:
     # Include emergency data storage (EMDS):
     # The models that uses the EMDS is already included, remove the models that
     # does not use EMDS.
     extra_configs:
```
Comment (m-alperen-sener) [inline tests/subsys/bluetooth/mesh/models/testcase.yaml:28]: the same here:
```suggestion
    extra_configs: OVERLAY_CONFIG=overlay-mesh-emds.conf
```

**Review by m-alperen-sener — CHANGES_REQUESTED** [review]

File: tests/subsys/bluetooth/mesh/models/prj.conf [hunk 3]
```diff
@@ -65,3 +65,5 @@ CONFIG_BT_MESH_LIGHT_SAT_SRV=y
 CONFIG_BT_MESH_LIGHT_HSL_SRV=y
 CONFIG_BT_MESH_LIGHT_HSL_CLI=y
 CONFIG_BT_MESH_SCHEDULER_CLI=y
+CONFIG_DM_MODULE=y
+CONFIG_BT_MESH_DM_CLI=y
```
Comment (m-alperen-sener) [inline tests/subsys/bluetooth/mesh/models/prj.conf:69]: Shouldn't we disable legacy sensor APIs here?

**Review by alxelax — CHANGES_REQUESTED** [review]
needs more work

**Comment by m-alperen-sener** [comment]
> warning: Deprecated symbol BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE is enabled.

Warnings are fine experimental; but,  "Deprecated symbol BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE is enabled." we should disable this.

On the other hand, we need to add app.overlay to silence actual build warning for dm-timer.
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/bluetooth/nrf_dm/app.overlay#L8-L11 
without dm-gpios section.

**Review by alxelax — APPROVED** [review]

File: tests/subsys/bluetooth/mesh/models/prj.conf (hunk 3 above)
Comment (akredalen) [reply tests/subsys/bluetooth/mesh/models/prj.conf:69]: I forgot to add this to my latest push. Is disabling this still necessary?

File: tests/subsys/bluetooth/mesh/models/app.overlay [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+ / {
+	/* The timer instance to use. */
+	chosen {
+		ncs,dm-timer = &timer2;
+	};
+
+	/* The selected pins will be used by Distance Measurement module for debug purposes. */
```
Comment (m-alperen-sener) [inline tests/subsys/bluetooth/mesh/models/app.overlay:13]: We don't need those pin entries here IMO.

File: tests/subsys/bluetooth/mesh/models/prj.conf (hunk 3 above)
Comment (m-alperen-sener) [reply tests/subsys/bluetooth/mesh/models/prj.conf:69]: We should stop using soon to be deprecated legacy API.

File: tests/subsys/bluetooth/mesh/models/app.overlay (hunk 4 above)
Comment (akredalen) [reply tests/subsys/bluetooth/mesh/models/app.overlay:13]: @PavelVPV

**Review by ludvigsj — APPROVED** [review]

File: tests/subsys/bluetooth/mesh/models/app.overlay (hunk 4 above)
Comment (PavelVPV) [reply tests/subsys/bluetooth/mesh/models/app.overlay:13]: Right, can be removed. Could you please also rename file to `dm.overlay` to reflect  that it belongs to distance measurement module?

**Comment by PavelVPV** [comment]
Added @nordicjm as requested when testcase.yml files change.

**Comment by nordicjm** [comment]
> Added @nordicjm as requested when testcase.yml files change.

Bookmarked, thanks

**Review by PavelVPV — APPROVED** [review]

**Comment by akredalen** [comment]
Forgot to disable legacy sensor API. Added now.

**Review by PavelVPV — APPROVED** [review]

**Pull request merged**